### DISCUSSION TITLE
NextJS <Script /> tag is not working with data-category (v3)

### DISCUSSION THREAD

**Discussion by dmendiboure** [orestbida/cookieconsent#580] [Q&A] [answered]
Hello,

I'm trying to implement `cookieconsent` in a NextJS 13 project. Everything seems to work fine, but when I try to block a script it doesn't work with the NextJS Script tag. Maybe I'm doing someting wrong.

Here's a CodeSandbox that illustrates the problem : https://codesandbox.io/p/sandbox/purple-paper-kw64mn?file=%2Fapp%2Fpage.tsx%3A18%2C36

**Comment by orestbida** [comment, marked as answer]
Hi @dmendiboure, for any React based framework I would suggest using [callback functions or event listeners](https://cookieconsent.orestbida.com/advanced/callbacks-events.html#callbacks-and-events). I'm not sure how the Nextjs Script tag works, so I can't help with that, sorry.

Here is a [demo with Next.js 13 on Stackblitz](https://stackblitz.com/edit/stackblitz-starters-gobdmp?file=components%2FCookieConsent.tsx,app%2Flayout.tsx%3AL20), using event listeners.

**Reply by pc-perrycima** [reply]
Your approach is absolutely correct, but I would like to know how you intend to activate/deactivate the Google Analytics script. I am using GoogleAnalytics from '@next/third-parties/google'. I cannot use the same logic with your example because I cannot understand how to activate the component only when the Analytics cookie is accepted.
